### ISSUE TITLE
Doesn't work with esbuild 0.24.0 ?

### ISSUE THREAD

**Issue by AFCMS** [imranbarbhuiya/esbuild-plugins-node-modules-polyfill#296]
### Issue description

I am trying to build a VSCode extension for the web, with polyfills for the node libraries I use. 

I have the following esbuild code:

```ts
/// <reference types="node" />

import esbuild from "esbuild";

import { nodeModulesPolyfillPlugin } from "esbuild-plugins-node-modules-polyfill";

const production = process.argv.includes("--production");

const esbuildProblemMatcherPlugin: esbuild.Plugin = {
    name: "esbuild-problem-matcher",

    setup(build) {
        build.onStart(() => {
            console.log("[esbuild] build started");
        });
        build.onEnd((result) => {
            result.errors.forEach(({ text, location }) => {
                console.error(`✘ [ERROR] ${text}`);
                if (location) {
                    console.error(`    ${location.file}:${location.line}:${location.column}:`);
                }
            });
            console.log("[esbuild] build finished");
        });
    },
};

async function main() {
    const sharedOptions: esbuild.BuildOptions = {
        entryPoints: ["src/extension.ts", "src/worker.ts"],
        bundle: true,
        target: "es2020",
        format: "cjs",
        external: ["vscode"],
        minify: production,
        sourcemap: !production,
        logLevel: "silent",
    };

    const mainOptions: esbuild.BuildOptions = {
        ...sharedOptions,
        platform: "node",
        outdir: "dist/main",
        plugins: [esbuildProblemMatcherPlugin],
    };

    const webOptions: esbuild.BuildOptions = {
        ...sharedOptions,
        platform: "browser",
        outdir: "dist/web",
        plugins: [
            nodeModulesPolyfillPlugin({
                modules: ["node:path", "node:worker_threads"],
            }),
            esbuildProblemMatcherPlugin,
        ],
    };

    const ctxMain = await esbuild.context(mainOptions);
    const ctxWeb = await esbuild.context(webOptions);

    await ctxMain.rebuild();
    await ctxMain.dispose();

    await ctxWeb.rebuild();
    await ctxWeb.dispose();
}

main().catch((e) => {
    console.error(e);
    process.exit(1);
});
```

But the two node packages are still not found for the web context:

```
Error: Build failed with 2 errors:
src/extension.ts:4:17: ERROR: Could not resolve "node:path"
src/extension.ts:7:23: ERROR: Could not resolve "node:worker_threads"
    at failureErrorWithLog (/home/afcm/dev/oxipng/node_modules/.pnpm/esbuild@0.24.0/node_modules/esbuild/lib/main.js:1476:15)
    at /home/afcm/dev/oxipng/node_modules/.pnpm/esbuild@0.24.0/node_modules/esbuild/lib/main.js:945:25
    at /home/afcm/dev/oxipng/node_modules/.pnpm/esbuild@0.24.0/node_modules/esbuild/lib/main.js:1354:9
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}
```

I don't know if I did something wrong here, it's a bit weird since the package depends on esbuild 0.24 but still specifies peer dependency to <=0.23, maybe it's unsupported yet?

```
Packages: +9
+++++++++
Progress: resolved 263, reused 207, downloaded 7, added 9, done

devDependencies:
+ esbuild-plugins-node-modules-polyfill 1.6.7

 WARN  Issues with peer dependencies found
.
└─┬ esbuild-plugins-node-modules-polyfill 1.6.7
  └── ✕ unmet peer esbuild@">=0.14.0 <=0.23.x": found 0.24.0
```

### Package version

1.6.7

### Node.js version

20.18.0

### Operating system

Fedora 41

**Comment by imranbarbhuiya** [comment]
Can you provide a mininal repro where I can test and fix?

**Comment by AFCMS** [comment]
[repro_esbuild.zip](https://github.com/user-attachments/files/17692062/repro_esbuild.zip)

**Comment by imranbarbhuiya** [comment]
currently [modules](https://github.com/imranbarbhuiya/esbuild-plugins-node-modules-polyfill#configure-which-modules-to-polyfill) expects name without `node:` prefix, so you can update your config to
```tsx
  const webOptions: esbuild.BuildOptions = {
    ...sharedOptions,
    platform: 'browser',
    outdir: 'dist/web',
    plugins: [
      nodeModulesPolyfillPlugin({
        modules: ['fs'],
      }),
      esbuildProblemMatcherPlugin,
    ],
  };
```

I'll work on a fix to allow both prefix and non-prefix one

**Comment by imranbarbhuiya** [comment]
Published v1.6.8 which fixes both peer deps issues and also allows `node:` prefix in modules
